File: Data Structures & Algorithms/contains-duplicate-ii/submission-1.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        
        #if k == 0:
        #    return True

        dup_dict = {}
        for ind,val in enumerate(nums):
            if val not in dup_dict:
                dup_dict[val] = ind
            else:
                if abs(dup_dict[val] - ind) <= k:
                    return True
                else:
                    dup_dict[val] = ind
        return False


        
            

File: Data Structures & Algorithms/maximum-product-subarray/submission-1.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {

        int max_prod = -10000;
        for (int i=0; i<nums.size(); i++)
        {
            int prod = 1;
            for (int j=i; j<nums.size();j++)
            {
                prod = prod * nums[j];
                if (prod > max_prod)
                {
                    max_prod = prod;
                }
            }
        }
        return max_prod;
    }
};
